runners/google-cloud-dataflow-java/examples-streaming/build.gradle

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import groovy.json.JsonOutput
+
+apply plugin: org.apache.beam.gradle.BeamModulePlugin
+applyJavaNature(failOnWarning: true, publish: false)
+// Evaluate the given project before this one, to allow referencing
+// "sourceSets.test.output" directly.
+evaluationDependsOn(":beam-examples-java")
+
+configurations { dataflowStreamingRunnerPreCommit }
+
+dependencies {
+  testRuntimeOnly project(path: ":beam-examples-java", configuration: "shadow")
+  testRuntimeOnly project(path: ":beam-examples-java", configuration: "shadowTest")
+  testRuntimeOnly project(path: ":beam-runners-google-cloud-dataflow-java", configuration: "shadow")
+}
+
+task preCommit(type: Test) {
```

Comment:
Gonna need bits like this:

```
 def gcsProject = project.findProperty('gcsProject') ?: 'apache-beam-testing'	 def gcsProject = project.findProperty('gcsProject') ?: 'apache-beam-testing'
 def gcsTempRoot = project.findProperty('gcsTempRoot') ?: 'gs://temp-storage-for-end-to-end-tests/'
```

It looks as though this whole block is actually pretty generic, so you could move it into the plugin and just specify the tests to include as a param.

Reply:
(I'm totally happy to merge it with duplication to speed things up, and then refactor later)